### ISSUE TITLE
[1337x] Exception (1337x): Connection reset by peer: The SSL connection could not be established, see inner exception. (Test)

### ISSUE THREAD

**Issue by mouzzampk2014** [Jackett/Jackett#4940]
Good Morning, 

Any idea how can I fix this. I tried changing different url but still same issue.

Thank you

**Jackett version**:  0.11.92.0

-------------------------------

**Comment by garfield69** [comment]
try upgrading to the latest jackett.

**Comment by mouzzampk2014** [comment]
Same issue :) 0.11.92.0

**Comment by garfield69** [comment]
0.11.107 is the latest version.
but if that is still not working for you, then answer a few more q:
what platform are you on?
are you using mono, if so what version?
are any of the other indexer working?
which domains have you tested for 1337x?

**Comment by mouzzampk2014** [comment]
All other working fine :) I tried few domains and working fine directly
from browser but under jackett failed and I'm on docker.

**Comment by garfield69** [comment]
then wait until docker puts out 0.11.107 and try again. report success or failure.

**Comment by mouzzampk2014** [comment]
Definitely and thank you so much :)

**Comment by garfield69** [comment]
and specify which 1337x domains are not working!

**Comment by mouzzampk2014** [comment]
Oops same issue and I tried all those ones 

`https://1337x.unblocked.lol 
https://1337x.st
https://1337x.unblockall.org
https://1337x.unblocker.cc
https://x1337x.ws/
https://1337x.unblocked.bet
https://filesdownloader.com
https://sitenable.ch`

Thank you

**Comment by mouzzampk2014** [comment]
And I am having issues adding IPTorrents 
![4](https://user-images.githubusercontent.com/44548525/54862746-13aaa000-4d37-11e9-9a89-32c4f32b9d12.JPG)

**Comment by garfield69** [comment]
yeah, lets not muddy the waters by bringing in iptorrents just now.

`https://1337x.unblocked.lol/` redirects to `https://1337x1.unblocked.lol/`
redirects break the 1337x indexer.
`https://1337x.unblocked.bet/` redirects to `https://1337x.unblocked.krd/`
`https://filesdownloader.com/` redirects to `http://sitenable.com/` which is a proxy service not the 1337x site.
and `https://sitenable.ch` redirects to `https://sitenable.asia/` and both are alias of the above proxy service.

the following 1337x domains work for me on my win10  Jackett 0.11.107 
```
https://1337x.st/
https://1337x1.unblocked.lol/ 
https://1337x.unblockall.org/
https://1337x.unblocker.cc/
https://x1337x.ws/
https://1337x.unblocked.krd/
```

so, from your docket jackett image, open a terminal and please post the output of 
`curl -vvv https://1337x.st/`
and 
`openssl s_client -connect 1337x.st:443`

**Comment by mouzzampk2014** [comment]
Morning, I am not sure how to open a terminal from my jackett image? :(

**Comment by garfield69** [comment]
I dont use docker so I don't know for sure, but browsing https://hub.docker.com/r/linuxserver/jackett/ i spotted this:
**Support Info**
Shell access whilst the container is running: `docker exec -it jackett /bin/bash`

**Comment by mouzzampk2014** [comment]
`

  | Jackett.Common.IndexerException: Exception (1337x): Connection reset by peer ---> System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception. ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer. ---> System.Net.Sockets.SocketException: Connection reset by peer    --- End of inner exception stack trace ---    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error)    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)    at System.Net.FixedSizeReader.ReadPacketAsync(Stream transport, AsyncProtocolRequest request)    at System.Net.Security.SslState.ThrowIfExceptional()    at System.Net.Security.SslState.InternalEndProcessAuthentication(LazyAsyncResult lazyResult)    at System.Net.Security.SslState.EndProcessAuthentication(IAsyncResult result)    at System.Net.Security.SslStream.EndAuthenticateAsClient(IAsyncResult asyncResult)    at System.Net.Security.SslStream.<>c.<AuthenticateAsClientAsync>b__47_1(IAsyncResult iar)    at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization) --- End of stack trace from previous location where exception was thrown ---    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)    --- End of inner exception stack trace ---    at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)    at System.Threading.Tasks.ValueTask`1.get_Result()    at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)    at System.Threading.Tasks.ValueTask`1.get_Result()    at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask)    at System.Threading.Tasks.ValueTask`1.get_Result()    at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)    at System.Net.Http.DecompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)    at CloudFlareUtilities.ClearanceHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)    at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)    at Jackett.Common.Utils.Clients.HttpWebClientNetCore.Run(WebRequest webRequest) in /home/appveyor/projects/jackett/src/Jackett.Common/Utils/Clients/HttpWebClientNetCore.cs:line 214    at Jackett.Common.Utils.Clients.WebClient.GetString(WebRequest request) in /home/appveyor/projects/jackett/src/Jackett.Common/Utils/Clients/WebClient.cs:line 114    at Jackett.Common.Indexers.BaseWebIndexer.RequestStringWithCookies(String url, String cookieOverride, String referer, Dictionary`2 headers) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 445    at Jackett.Common.Indexers.CardigannIndexer.PerformQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/CardigannIndexer.cs:line 1229    at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 331    --- End of inner exception stack trace ---    at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 351    at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery(TorznabQuery query) in /home/appveyor/projects/jackett/src/Jackett.Common/Indexers/BaseIndexer.cs:line 815    at Jackett.Common.Services.IndexerManagerService.TestIndexer(String name) in /home/appveyor/projects/jackett/src/Jackett.Common/Services/IndexerManagerService.cs:line 222    at Jackett.Server.Controllers.IndexerApiController.Test() in /home/appveyor/projects/jackett/src/Jackett.Server/Controllers/IndexerApiController.cs:line 128    at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)    at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()    at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()    at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)    at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)    at Microsoft.AspNetCore.Builder.Extensions.UsePathBaseMiddleware.Invoke(HttpContext context)    at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/appveyor/projects/jackett/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 29
-- | --`

**Comment by mouzzampk2014** [comment]
That's the one?

**Comment by garfield69** [comment]
no. look I don't know enough about docket to guide you through this, so I'm going to ask @flightlevel  if he can assist with this ticket. Maybe he can help. Thanks.

**Comment by mouzzampk2014** [comment]
> no. look I don't know enough about docket to guide you through this, so I'm going to ask @flightlevel  if he can assist with this ticket. Maybe he can help. Thanks.

Its perfectly fine. Please enjoy your weekend. Thanks again

**Comment by flightlevel** [comment]
@mouzzampk2014 Are you using the linux server docker image?
From a terminal run `docker exec -it jackett /bin/bash` replace `jackett` with whatever you called your Jackett container. You now have bash access to container
run these two commands and post the output
`curl -vvv https://1337x.st/`
and
`openssl s_client -connect 1337x.st:443`

**Comment by mouzzampk2014** [comment]
> @mouzzampk2014 Are you using the linux server docker image?
> From a terminal run docker exec -it jackett /bin/bash replace jackett with whatever you called your Jackett container. You now have bash access to container
> run these two commands and post the output
> curl -vvv https://1337x.st/
> and
> openssl s_client -connect 1337x.st:443

![3](https://user-images.githubusercontent.com/44548525/54876848-b97d0e00-4e0e-11e9-890a-27e42b2c44fe.JPG)
![5](https://user-images.githubusercontent.com/44548525/54876849-b97d0e00-4e0e-11e9-962b-6bc48dcab60f.JPG)

Thank you :)

**Comment by flightlevel** [comment]
and try `curl -vvv https://www.google.com/` from the container as well

**Comment by mouzzampk2014** [comment]
> and try curl -vvv https://www.google.com/ from the container as well

![2](https://user-images.githubusercontent.com/44548525/54884594-e5c67800-4e6a-11e9-929f-2b076995b456.JPG)
![3](https://user-images.githubusercontent.com/44548525/54884595-e5c67800-4e6a-11e9-8658-0a44b78f5991.JPG)

**Comment by garfield69** [comment]
so the connect to google com via https works, but the one to 1337x.st does not.
it is looking like your access to 1337x.st is being blocked by something and redirecting to a black hole  that does not have the requested certificate.
here is a  bunch of 1337 domains you can try with the curl command, see if any of them return something more like the google response than the SSL_ERROR_SYSCALL of 1337x.st
```
curl -vvv https://1337x.to/
curl -vvv https://x1337x.ws/
curl -vvv https://x1337x.eu/
curl -vvv https://x1337x.se/
curl -vvv https://1337x.is/
```

**Comment by mouzzampk2014** [comment]
> so the connect to google com via https works, but the one to 1337x.st does not.
> it is looking like your access to 1337x.st is being blocked by something and redirecting to a black hole  that does not have the requested certificate.
> here is a  bunch of 1337 domains you can try with the curl command, see if any of them return something more like the google response than the SSL_ERROR_SYSCALL of 1337x.st
> curl -vvv https://1337x.to/
> curl -vvv https://x1337x.ws/
> curl -vvv https://x1337x.eu/
> curl -vvv https://x1337x.se/
> curl -vvv https://1337x.is/

Same issue

![2](https://user-images.githubusercontent.com/44548525/55053385-ecfcaa00-5053-11e9-8d2d-237e535bda5f.JPG)

![3](https://user-images.githubusercontent.com/44548525/55053386-ecfcaa00-5053-11e9-9ac5-6e377a9d4b52.JPG)

![4](https://user-images.githubusercontent.com/44548525/55053387-ecfcaa00-5053-11e9-8c24-c9ad4b8bcd97.JPG)

![5](https://user-images.githubusercontent.com/44548525/55053388-ecfcaa00-5053-11e9-8978-8b5e5e090ad9.JPG)

![6](https://user-images.githubusercontent.com/44548525/55053389-ed954080-5053-11e9-8e19-d1bff50f67ef.JPG)

**Comment by garfield69** [comment]
are you using the DNS provided by your ISP?
if so then consider (as a test, or permanently if this works) switching to either the cloudflare DNS at 1.1.1.1 an 1.0.0.1 or Googles at 8.8.8.8 and 8.8.4.4
and see if that makes any difference.

**Comment by mouzzampk2014** [comment]
Hi,

I'm using reverse proxy and set up with letsencrypt, cloudflare and nginx.

**Comment by garfield69** [comment]
Oh, well that's just too complicated for me. @flightlevel  @kaso17  I'm passing this one to you, I'm out of ideas on this one.

**Comment by flightlevel** [comment]
Closing this issue as it has nothing to do with Jackett and troubleshooting each individuals network configuration is beyond the Jackett team.
If you can't reach a site using a common command line tool like curl there is no chance Jackett will work, it indicates that there is some sort of connection/setup/network/configuration issue unrelated to Jackett

**Comment by mouzzampk2014** [comment]
You are right I have 15 other private and they are connecting properly and only one public giving problems :)

**Comment by genrl** [comment]
I found a similar issue with setting up Torrentleech and just posting here incase it helps someone down the line. I have a raspberry pi with OpenVPN and AirVPN running but for some reason Jackett (latest ver as a service), Sonarr and Radarr were bypassing the VPN. As I'm in Australia, our ISPs block certain content therefore calls to someone of the indexers were being blocked and I was gettting the above error. Initially I tried to forward all traffic through the VPN using iptables ([here](https://berndknows.blogspot.com/2013/11/configure-raspberry-pi-as-openvpn-client.html) and [here)](https://unix.stackexchange.com/questions/283801/iptables-forward-traffic-to-vpn-tunnel-if-open) but this also didn't seem to work (although I confirmed 
qBittorrent was using the VPN thankfully ([using this](https://www.vpnuniversity.com/bittorrent/how-to-check-your-torrent-ip-address)). Anyway in the end I decded to change the DNS on my router to Cloudfare (1.1.1.1 and 1.0.0.1) - which Jackett seemed to bypass again - but I then applied the same settings on my raspberry pi [using this](https://pimylifeup.com/raspberry-pi-dns-settings/) and it's working now. Altering the guides above on the curl were helpful in troubleshooting the error - thanks all and sorry for adding to old thread.
